src/app/extensions/copilot/store/copilot-config/copilot-config.effects.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Store, select } from '@ngrx/store';
+import { map, switchMap } from 'rxjs/operators';
+
+import { whenFalsy } from 'ish-core/utils/operators';
+import { StatePropertiesService } from 'ish-core/utils/state-transfer/state-properties.service';
+
+import { CopilotConfig } from '../../models/copilot-config/copilot-config.model';
+
+import { loadCopilotConfig, setCopilotConfig } from './copilot-config.actions';
+import { getCopilotConfig } from './copilot-config.selectors';
+
+@Injectable()
+export class CopilotConfigEffects {
+  constructor(
+    private actions$: Actions,
+    private store: Store,
+    private statePropertiesService: StatePropertiesService
+  ) {}
+
+  loadCopilotConfiguration$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(loadCopilotConfig),
+      switchMap(() =>
+        this.statePropertiesService
+          .getStateOrEnvOrDefault<CopilotConfig>('COPILOT', 'copilot')
+          .pipe(map(config => setCopilotConfig({ config })))
```

Comment:
somehow the setCopilotConfig action is triggered twice, adding distinctUntilChanged() operator should fix it